pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```diff
@@ -112,6 +116,22 @@ func ensureNodeKubeconfigs(ctx context.Context, client coreclientv1.CoreV1Interf
 		return fmt.Errorf("system:admin client private key missing from secret %s/node-system-admin-client", operatorclient.OperatorNamespace)
 	}
 
+	// Ensure secret key matches the certificate
+	_, err = tls.X509KeyPair(systemAdminClientCert, systemAdminClientKey)
```

Comment:
could you point me to a location where this cert is created?

Reply:
We could read it from the secret, but that would require certrotationcontroller to set those first - I'm not entirely sure we can rely on that

Reply:
sure, updated